### PR TITLE
Add Bento opt-in

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

To opt-in to serving the Bento versions of AMP components (where available), do:

```php
add_filter( 'amp_bento_enabled', '__return_true' );
```

Note that this includes the Bento AMP experiment opt-in script, so it will force dev mode as well.

See #6443.

- [x] Add filter to opt-in to Bento mode.
- [x] Enable [bento experiment](https://amp.dev/documentation/guides-and-tutorials/start/bento_guide/).
- [x] Remove `amp` attribute from `html` and add `data-ampdevmode` to Bento experiment opt-in `script`. Done in https://github.com/ampproject/amp-wp/pull/6518.
- [x] ~The image sanitizer no longer converts `img` to `amp-img` but instead just ensures dimensions are provided.~ See https://github.com/ampproject/amp-wp/pull/6518.
- [x] Bento versions of scripts are used instead of `latest`.
- [x] ~The `amp-runtime` is omitted as a dependency of any Bento scripts.~ (Not yet supported by AMP.)
- [x] ~Bento stylesheets are enqueued for any Bento components used on the page. (Should these be excluded from CSS concat?)~ (Not needed when part of AMP page.)
- [x] Disable Optimizer SSR? See https://github.com/ampproject/amphtml/issues/35485.
- [x] ~Skip converting any non-Bento components?~ (Not relevant in the context of this PR.)

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
Note to self: after #6530 is merged, rebase into `develop` and cherry-pick `add/bento-with-amphtml-spec-20210811-spec-update`.

File: includes/amp-helper-functions.php [hunk 1]
```diff
@@ -964,6 +997,28 @@ function amp_register_default_styles( WP_Styles $styles ) {
 		AMP__VERSION
 	);
 	$styles->add_data( 'amp-icons', 'rtl', 'replace' );
+
+	// These are registered exclusively for used for non-AMP pages that manually enqueue them. They aren't needed on
```
Comment (pierlon) [inline includes/amp-helper-functions.php]:
```suggestion
	// These are registered exclusively for non-AMP pages that manually enqueue them. They aren't needed on
```

**Review by pierlon — APPROVED** [review]

**Comment by pierlon** [comment]
Would you update the `prefer_bento` arg for `AMP_Tag_And_Attribute_Sanitizer` to use `amp_is_bento_enabled()` as well?

**Comment by westonruter** [comment]
> Would you update the `prefer_bento` arg for `AMP_Tag_And_Attribute_Sanitizer` to use `amp_is_bento_enabled()` as well?

Oh yes, indeed.

**Comment by milindmore22** [comment]
**QA Passed**

✅  Add filter to opt-in to Bento mode.

✅   Enable bento experiment.

✅  Bento versions of scripts are used instead of the latest. bento scripts are converted to amp scripts 1.0 version

⚠️  Disable Optimizer SSR?

**Comment by westonruter** [comment]
> ⚠️ Disable Optimizer SSR?

This one is obsolete. Bento is compatible with Optimized AMP now.

**Pull request merged**